internal/controller/nonadminbackup_controller.go

```diff
@@ -113,7 +385,12 @@ func (r *NonAdminBackupReconciler) Reconcile(ctx context.Context, req ctrl.Reque
 // If it is empty, it sets the Phase to "New".
 // It then returns boolean values indicating whether the reconciliation loop should requeue or exit
 // and error value whether the status was updated successfully.
-func (r *NonAdminBackupReconciler) init(ctx context.Context, logger logr.Logger, nab *nacv1alpha1.NonAdminBackup) (bool, error) {
+func (r *NonAdminBackupReconciler) initNabCreate(ctx context.Context, logger logr.Logger, nab *nacv1alpha1.NonAdminBackup) (bool, error) {
+	// If the NonAdminBackup is marked for deletion (either by DeleteBackup in the spec or with a DeletionTimestamp), skip initNabCreate
+	if nab.Spec.DeleteBackup || !nab.ObjectMeta.DeletionTimestamp.IsZero() {
```

Comment:
please remove all this checks in the following functions

Reconcile should not reach this stage of reconcile if it is marked for deletion

Reply:
But this will continue to enter those functions, right?

we do not have `EXIT NOW RECONCILE` return statement, so having `return false, nil` will continue to the next reconcile function and if it's marked as delete it will enter them?